### PR TITLE
octopus: rgw/notification: support GetTopicAttributes API

### PR DESCRIPTION
**Pull request by smithfarm**
backport tracker: https://tracker.ceph.com/issues/48462

---

backport of https://github.com/ceph/ceph/pull/38171
parent tracker: https://tracker.ceph.com/issues/46296

this backport was staged using ceph-backport.sh version 16.0.0.6848
find the latest version at https://github.com/ceph/ceph/blob/master/src/script/ceph-backport.sh

### CODE REVIEW

**Comment by smithfarm** [comment]
```
/home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_pubsub.cc: In member function 'std::string rgw_pubsub_sub_dest::to_json_str() const':
/home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_pubsub.cc:405:29: error: 'persistent' was not declared in this scope
  405 |   encode_json("Persistent", persistent, &f);
      |                             ^~~~~~~~~~
make[3]: *** [src/rgw/CMakeFiles/rgw_common.dir/build.make:860: src/rgw/CMakeFiles/rgw_common.dir/rgw_pubsub.cc.o] Error 1
```

File: src/rgw/rgw_pubsub.cc [hunk 1]
```diff
@@ -372,6 +392,23 @@ void rgw_pubsub_sub_dest::dump_xml(Formatter *f) const
...
+{
+  // first 2 members are omitted here since they
+  // dont apply to AWS compliant topics
+  JSONFormatter f;
+  f.open_object_section("");
+  encode_json("EndpointAddress", push_endpoint, &f);
+  encode_json("EndpointArgs", push_endpoint_args, &f);
+  encode_json("EndpointTopic", arn_topic, &f);
+  encode_json("HasStoredSecret", stored_secret, &f);
+  encode_json("Persistent", persistent, &f);
```
Comment (smithfarm) [inline src/rgw/rgw_pubsub.cc:405]:
```
/home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_pubsub.cc: In member function 'std::string rgw_pubsub_sub_dest::to_json_str() const':
/home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_pubsub.cc:405:29: error: 'persistent' was not declared in this scope
  405 |   encode_json("Persistent", persistent, &f);
      |                             ^~~~~~~~~~
make[3]: *** [src/rgw/CMakeFiles/rgw_common.dir/build.make:860: src/rgw/CMakeFiles/rgw_common.dir/rgw_pubsub.cc.o] Error 1
```

**Comment by smithfarm** [comment]
non-trivial to backport

**Comment by yuvalif** [comment]
created: https://github.com/ceph/ceph/pull/40812 to resolve the above conflicts

**Pull request closed without merging**